main.go

```diff
@@ -1,9 +1,74 @@
 package main
 
```

Comment:
this looks ok, but I am going to move it all to the right place later tonight

Reply:
Ah alright